citadel/indico_citadel/util.py

```diff
@@ -215,3 +215,14 @@ def get_user_access(user, admin_override_enabled=False):
                             for x in user.iter_all_multipass_groups()]
         access += _include_capitalized_groups(multipass_groups)
     return access
+
+def check_event_categories(category):
+        from indico_citadel.plugin import CitadelPlugin
+        from indico.modules.events import Event
+
+        threshold = CitadelPlugin.settings.get('large_category_warning_threshold')
+        num_events = (Event.query.with_parent(category).count())
+
+        if threshold and num_events > threshold:
+            # TODO: Translate + Better message (e.g. recommend 'groups' instead of 'categories')
+            return f"The category '{category.title}' contains {num_events} events, which may cause performance issues. "
```

Comment:
Even when just using the signal you can `render_plugin_template` for this. That way you can also nicely wrap it in the `message_box` macro.